lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -1415,9 +1415,14 @@ def _hide(self, item: HistoryItemModel):
     def _unhide(self, item: HistoryItemModel):
         item.visible = True
 
-    def _delete(self, item: HistoryItemModel):
-        manager = self._get_item_manager(item)
-        manager.delete(item, flush=self.flush)
+    def _delete(self, item: HistoryItemModel, trans: ProvidesHistoryContext):
+        if getattr(item, "deleted", False):
+            # TODO: remove this `update` when we can properly track the operation results to notify the history
```

Comment:
I don't know if I understand the TODO 😅. We get here if the item is already deleted, so it seems like we wouldn't need to do anything here ?

Reply:
You are absolutely right, too much copy&paste :sweat_smile: this was only necessary for the collection `purge` as there, the purge was not triggering the change in the history update time making the operation seem to be running until something else bumped the history update (if I recall correctly).

Thanks for the catch, I'll remove it.